### PR TITLE
Add opset 15 kernels for Pow

### PR DESCRIPTION
**Pull request by hariharans29**
**Description**: Pow had an opset bump with the inclusion of `bfloat16` in the list of supported types for the exponent input. 

**Motivation and Context**
ONNX compliance

### CODE REVIEW

File: onnxruntime/core/providers/cpu/nn/batch_norm.cc [hunk 1]
```diff
@@ -31,18 +31,44 @@ ONNX_CPU_OPERATOR_VERSIONED_TYPED_KERNEL(BatchNormalization, 7, 8, double,
...
+                                             .TypeConstraint("T", DataTypeImpl::GetTensorType<float>())
+                                             .TypeConstraint("U", DataTypeImpl::GetTensorType<float>()),
+                                         BatchNorm<float>);
+
+ONNX_CPU_OPERATOR_VERSIONED_TYPED_KERNEL(BatchNormalization, 14, 14, double,
+                                         KernelDefBuilder()
+                                             .Alias(3, 1)
+                                             .Alias(4, 2)
+                                             .TypeConstraint("T", DataTypeImpl::GetTensorType<double>())
+                                             .TypeConstraint("U", DataTypeImpl::GetTensorType<double>()),
```
Comment (hariharans29) [inline onnxruntime/core/providers/cpu/nn/batch_norm.cc:54]: Bug fix: "U" was missing for opset-14

File: onnxruntime/core/providers/cuda/nn/batch_norm.cc [hunk 2]
```diff
@@ -11,33 +11,45 @@ using namespace std;
...
+      BatchNorm<T>);                                               \
+  ONNX_OPERATOR_VERSIONED_TYPED_KERNEL_EX(                         \
+      BatchNormalization,                                          \
+      kOnnxDomain,                                                 \
+      14, 14,                                                      \
+      T,                                                           \
+      kCudaExecutionProvider,                                      \
+      (*KernelDefBuilder::Create())                                \
+          .TypeConstraint("T", DataTypeImpl::GetTensorType<T>())   \
+          .TypeConstraint("U", DataTypeImpl::GetTensorType<T>()),  \
```
Comment (hariharans29) [inline onnxruntime/core/providers/cuda/nn/batch_norm.cc:41]: "U" was missing for opset-14 (https://github.com/microsoft/onnxruntime/pull/7806/files#r673442347)

File: onnxruntime/core/optimizer/shape_to_initializer.cc [hunk 3]
```diff
@@ -1,80 +0,0 @@
...
-#include "core/optimizer/initializer.h"
-#include "core/optimizer/optimizer_execution_frame.h"
-#include "core/framework/op_kernel.h"
-#include "core/framework/tensorprotoutils.h"
-
-namespace onnxruntime {
-
-Status ShapeToInitializer::Apply(Graph& graph, Node& node, RewriteRuleEffect& rule_effect, const logging::Logger&) const {
-  // Store the statically inferred shape of the input to the Shape operator.
-  const ONNX_NAMESPACE::TensorShapeProto* input_shape_proto = node.InputDefs()[0]->Shape();
```
Comment (hariharans29) [inline onnxruntime/core/optimizer/shape_to_initializer.cc:17]: Retiring this RewriteRule as its functionality is already offered by another GraphTransformer (ConstantFolding) at the same optimization level (Level 1). Updating this would have been a copy/paste of the new logic added to the Shape processing logic in ConstantFolding

File: onnxruntime/test/providers/cpu/nn/batch_norm_op_test.cc [hunk 4]
```diff
@@ -790,5 +790,31 @@ TEST(BatchNormTest, ForwardTrainingTestOpset14) {
   test.Run(OpTester::ExpectResult::kExpectSuccess, "", {kCudaExecutionProvider, kTensorrtExecutionProvider, kOpenVINOExecutionProvider});
 }
 
+TEST(BatchNormTest, ForwardTrainingTestOpset15) {
+  OpTester test("BatchNormalization", 15);
+  float epsilon = 1e-05f;
+  float momentum = 0.1f;
+  int64_t training_mode = 1;
```
Comment (hariharans29) [inline onnxruntime/test/providers/cpu/nn/batch_norm_op_test.cc:797]: trainig mode is enabled. But it still tests the opset 15 kernel's availability.

File: onnxruntime/test/testdata/onnx_backend_test_series_filters.jsonc [hunk 5]
```diff
@@ -67,7 +67,6 @@
         "^test_add_uint8_cpu",
         "^test_div_uint8_cpu",
         // Following tests are for opset 15 ops and are not yet implemented in ORT
-        "^test_shape_*",
```
Comment (hariharans29) [inline onnxruntime/test/testdata/onnx_backend_test_series_filters.jsonc:70]: Pow and BatchNorm came with no ONNX tests. Only Shape came with tests.

**Comment by pranavsharma** [comment]
LGTM 👍 . Can you resolve the pipelines?

**Comment by hariharans29** [comment]
> LGTM 👍 . Can you resolve the pipelines?

Yeah, it just needs the `OperatorKernels.md` file checked-in. Other failures are just due to flakiness. I will check the file in while trying to resolve comments from others (if any).

**Comment by hariharans29** [comment]
> LGTM 👍 . Can you resolve the pipelines?

Fixed the pipelines.

**Review by pranavsharma — APPROVED** [review]
